lib/host.js

```diff
@@ -251,8 +251,8 @@ Host.prototype.checkHealth = function (connection) {
  * @internal
  * @ignore
  */
-Host.prototype.removeFromPool = function (connection) {
-  this.pool.remove(connection);
+Host.prototype.removeFromPool = function (connection, cb) {
```

Comment:
Needed to add a callback here to detect when the connection is closed.  Not quite sure if my use of this is correct of if there is a better way (I want to remove the control connection from pool and create a new one when it's done).